src/value.rs

```diff
@@ -203,6 +204,61 @@ impl<'tcx> PrimVal {
             _ => Err(EvalError::InvalidBool),
         }
     }
+
+    pub fn signed_offset(self, i: i64, layout: &TargetDataLayout) -> EvalResult<'tcx, Self> {
+        match self {
+            PrimVal::Bytes(b) if b as u64 as u128 == b => Ok(PrimVal::Bytes(signed_offset(b as u64, i, layout)? as u128)),
+            PrimVal::Bytes(_) => Err(EvalError::OverflowingMath),
+            PrimVal::Ptr(ptr) => ptr.signed_offset(i, layout).map(PrimVal::Ptr),
+            PrimVal::Undef => Err(EvalError::ReadUndefBytes),
+        }
+    }
+
+    pub fn offset(self, i: u64, layout: &TargetDataLayout) -> EvalResult<'tcx, Self> {
+        match self {
+            PrimVal::Bytes(b) if b as u64 as u128 == b => Ok(PrimVal::Bytes(offset(b as u64, i, layout)? as u128)),
+            PrimVal::Bytes(_) => Err(EvalError::OverflowingMath),
+            PrimVal::Ptr(ptr) => ptr.offset(i, layout).map(PrimVal::Ptr),
+            PrimVal::Undef => Err(EvalError::ReadUndefBytes),
+        }
+    }
+
+    pub fn wrapping_signed_offset(self, i: i64, layout: &TargetDataLayout) -> EvalResult<'tcx, Self> {
+        match self {
+            PrimVal::Bytes(b) if b as u64 as u128 == b => Ok(PrimVal::Bytes(wrapping_signed_offset(b as u64, i, layout) as u128)),
+            PrimVal::Bytes(_) => Err(EvalError::OverflowingMath),
```

Comment:
I think the condition in the first branch should be `assert_eq!(b as u64 as u128, b)` because it's a *misuse* of this method on a `PrimVal::Bytes` that *is not* a pointer.